Menu/Footer: Fixed/updated a few link destinations.

EricDunsworth opened the pull request:
FYI some of these changes will need to be replicated in Canada.ca's real menu/footer.

CC @rlambert27 @shawnthompson @nicolasraymond

shawnthompson commented:
CC @lucas-hay @MaximMartel can you make sure this gets updated on Canada.ca also?

RobJohnston commented on site/data/i18n/en.json:
```diff
@@ -186,13 +186,13 @@
 	"menu-footer-about-military-text": "Public service and military",
 	"menu-footer-about-military-link": "https://www.canada.ca/en/government/publicservice.html",
 	"menu-footer-about-news-text": "News",
-	"menu-footer-about-news-link": "http://news.gc.ca/",
```
This needs to be updated yet again.  Going to news.gc.ca redirects to news.gc.ca/web/index-en.do, but then there's a modal alert that says the the page has been moved:

> You will now be redirected to the Newsroom of Canada.ca.
> If you are not redirected in the next 20 seconds, please visit https://www.canada.ca/en/news.html

EricDunsworth replied:
I'll update it in a sec. Was working at the time I sent this PR. My change on this line is also causing GitHub's merge conflict warning.

A few days after I opened this PR, #1213 was opened and merged. It changed the link in question to point to http://www.servicecanada.gc.ca/gcnews. But since that's an unrelated domain for a site that was recently migrated into Canada.ca, IMO it'd be more ideal to make the link go directly to the news site.

nschonni commented on site/data/i18n/en.json:
```diff
@@ -30,7 +30,7 @@
 	"menu-main-benefits-id-main": "benefits",
 	"menu-main-health-text-main": "Health",
-	"menu-main-health-link-main": "https://www.canada.ca/en/services/health.html",
+	"menu-main-health-link-main": "http://healthycanadians.gc.ca/index-eng.php",
```
The old one is where this redirects

kodecount replied:
Most of `healthycanadians` are redirects.

EricDunsworth replied:
Sorry :(. Looks like I messed up my rebase/conflict resolutions.

I was under the mistaken impression that the only changes that had been made to these files since this PR was opened had come from #1213. So while rebasing, I re-used my versions of those files as-is without any second thought.

I'll sort it out soon.

nschonni commented on site/data/i18n/en.json:
```diff
@@ -96,8 +96,8 @@
-	"menu-main-business-text-4": "Doing business with government",
-	"menu-main-business-link-4": "https://www.canada.ca/en/services/business/doing-business.html",
+	"menu-main-business-text-4": "Sell to government",
+	"menu-main-business-link-4": "https://www.canada.ca/en/services/business/sell.html",
```
This redirects to the other one

nschonni commented on site/data/i18n/en.json:
```diff
@@ -186,13 +186,13 @@
 	"menu-footer-about-reporting-link": "https://www.canada.ca/en/transparency/reporting.html",
 	"menu-footer-about-pm-text": "Prime Minister",
-	"menu-footer-about-pm-link": "http://pm.gc.ca/eng",
+	"menu-footer-about-pm-link": "https://pm.gc.ca/eng",
```
Although the splash page appears to support HTTPS, the language specific home pages do not

kodecount commented on site/data/i18n/fr.json:
```diff
@@ -129,23 +130,23 @@
+	"menu-main-health-link-8": "http://canadiensensante.gc.ca/health-system-systeme-sante/index-fra.php",
 	"menu-main-health-text-9": "Science, recherche et données",
-	"menu-main-health-link-9": "https://www.canada.ca/fr/services/sante/science-recherche-et-donnees.html",
+	"menu-main-health-link-9": "http://canadiensensante.gc.ca/science-research-sciences-recherches/index-fra.php",
```
All of the `canadiensensante.gc.ca` seem to redirect to the old link.

EricDunsworth commented:
This PR should be good to go now.

I've fixed all the links updates from #1217 and #1222 that I mistakenly reverted in my first rebase. Also reverted the "menu-footer-about-pm-link" variables' values and fixed a few more links that have become broken over the past few months.
